drivers/console/native_posix_console.c

```diff
@@ -27,9 +39,199 @@ static int printk_init(struct device *arg)
 
 	extern void __printk_hook_install(int (*fn)(int));
 	__printk_hook_install(putchar);
+}
+
+/**
+ * Ensure that whatever was written thru printk is displayed now
+ */
+void posix_flush_stdout(void)
+{
+	fflush(stdout);
+}
+#endif /* CONFIG_NATIVE_POSIX_STDOUT_CONSOLE */
+
+#if defined(CONFIG_NATIVE_POSIX_STDIN_CONSOLE)
+
+static struct k_fifo *avail_queue;
+static struct k_fifo *lines_queue;
+static u8_t (*completion_cb)(char *line, u8_t len);
+
+static K_THREAD_STACK_DEFINE(stack, CONFIG_ARCH_POSIX_RECOMMENDED_STACK_SIZE);
+static struct k_thread native_stdio_thread;
+
+
+static inline void found_eof(void)
+{
+	/*
+	 * Once stdin is closed
+	 * or the input file has ended, there is no need to
+	 * try again
+	 */
+	ECHO("Got EOF\n");
+	k_thread_abort(&native_stdio_thread);
+}
+
+static int handle_native_shell_cmd(const char *s, u32_t *towait)
+{
+
+	if (strncmp(s, "!wait", 5) == 0) {
+		int ret;
+
+		ret = sscanf(&s[5], "%u", towait);
+		if (ret != 1) {
+			posix_print_error_and_exit("%s(): %s not understood, "
+					"syntax = !wait %%u\n",
+					__func__, s);
+		}
+		return 1;
+	}
+
+	posix_print_error_and_exit("%s(): '%s' not understood\n",
+				__func__, s);
+
+	return 0;
+}
+
+/**
+ * Check if there is any line in the stdin buffer,
+ * if there is and we have available shell buffers feed it to the shell
+ *
+ * This function returns how long the thread should wait in ms,
+ * before checking again the stdin buffer
+ */
+static u32_t attempt_read_from_stdin(void)
+{
+	struct timeval timeout;
+	fd_set readfds;
+	static struct console_input *cmd;
+	u32_t towait = CONFIG_NATIVE_STDIN_POLL_PERIOD;
```

Comment:
Note that Zephyr timeouts are s32_t (e.g. the type that k_sleep expects) since K_FOREVER == -1. Is u32_t intentionally chosen since you don't need K_FOREVER?